src/FakeItEasy/Configuration/FakeConfigurationManager.cs

```diff
@@ -61,6 +59,123 @@ public IAnyCallConfigurationWithNoReturnTypeSpecified CallTo(object fakeObject)
             return this.configurationFactory.CreateAnyCallConfiguration(manager, rule);
         }
 
+        public IPropertySetterConfiguration<TValue> CallToSet<TValue>(Expression<Func<TValue>> propertySpecification)
+        {
+            Guard.AgainstNull(propertySpecification, nameof(propertySpecification));
+            this.AssertThatMemberCanBeIntercepted(propertySpecification);
+
+            var setterExpression = BuildSetterFromGetter(propertySpecification);
+
+            return new PropertySetterConfiguration<TValue>(setterExpression, this.CreateVoidArgumentValidationConfiguration);
+        }
+
+        private static string GetIndexerName(MethodCallExpression methodCallExpression)
+        {
+            if (methodCallExpression != null &&
+                HasThis(methodCallExpression) &&
+                methodCallExpression.Method.IsSpecialName)
+            {
+                var methodName = methodCallExpression.Method.Name;
+                if (methodName.StartsWith("get_", StringComparison.Ordinal))
+                {
+                    return methodName.Substring(4);
+                }
+            }
+
+            return null;
+        }
+
+        private static bool HasThis(MethodCallExpression methodCallExpression)
+        {
+            return (methodCallExpression.Method.CallingConvention & CallingConventions.HasThis) ==

```

Comment:
Ah, yes, forgot about 3.5. Adding a *generic* extension method for enums is indeed tricky, since you can't use `Enum` as a generic constraint, but you can still create an extension method on the `Enum` type itself. However in this case it's probably not worth the trouble, because the implementation would have to depend on the underlying type.